mwim.el

```diff
@@ -125,37 +125,61 @@ to the end of line."
   (skip-chars-backward " \t"))
 
 ;;;###autoload
-(defun mwim-beginning-of-code-or-line ()
+(defun mwim-beginning-of-code-or-line (arg)
   "Move point to the beginning of code.
-If the point is already there, move to the beginning of line."
-  (interactive "^")
+If the point is already there, move to the beginning of line.
+
+If ARG is not nil, move forward ARG lines first. If point reaches
+the beginning or end of the buffer, stop there."
+  (interactive "^P")
```

Comment:
I think "P" does not suit here: for example, when I use `M--` (or `C--`)
prefix, I want to move one line backward; or `C-u` should move 4 lines
forward – these are the standard meanings for prefixes.  As you noted,
"p" is also not good.  That's why I prefer my solution.  Did you look at
it?